kubernetes/linux/setup.sh

```diff
@@ -12,12 +12,18 @@ fi
 sudo tdnf install ca-certificates-microsoft -y
 sudo update-ca-trust
 
-sudo tdnf install ruby-3.1.3 -y
+# sudo tdnf install ruby-3.1.3 -y
+tdnf install -y gcc patch bzip2 openssl-devel libyaml-devel libffi-devel readline-devel zlib-devel gdbm-devel ncurses-devel
+wget https://github.com/rbenv/ruby-build/archive/refs/tags/v20230330.tar.gz -O ruby-build.tar.gz
+tar -xzf ruby-build.tar.gz
+PREFIX=/usr/local ./ruby-build-*/install.sh
+ruby-build 3.1.3 /usr
+
 # remove unused default gem openssl, find as they have some known vulns
-rm /usr/lib/ruby/gems/3.1.0/specifications/default/openssl-3.0.1.gemspec
-rm -rf /usr/lib/ruby/gems/3.1.0/gems/openssl-3.0.1
-rm /usr/lib/ruby/gems/3.1.0/specifications/default/find-0.1.1.gemspec
-rm -rf /usr/lib/ruby/gems/3.1.0/gems/find-0.1.1
+# rm /usr/lib/ruby/gems/3.1.0/specifications/default/openssl-3.0.1.gemspec
```

Comment:
why these have been commented out?

Reply:
I was just checking the location of where these are downloaded....Have updated